### PR TITLE
Edits for CDN 2021 chapter

### LINKED ISSUE #2159: CDN 2021
# Part IV Chapter 21: CDN

CDN illustration

If you're interested in contributing to the CDN chapter of the 2021 Web Almanac, please reply to this issue and indicate which role or roles best fit your interest and availability: [author](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Authors'-Guide), [reviewer](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Reviewers'-Guide), [analyst](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Analysts'-Guide), and/or [editor](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Editors'-Guide).

## Content team

| Lead | Authors | Reviewers | Analysts | Editors | Coordinator |
| --- | ------- | --------- | -------- | ------- | -- |
| @Navaneeth-akam | @Navaneeth-akam | @boosef @jzyang @keane234 | @paulcalvano | @jzyang | @rviscomi |

<details>
<summary><b>Expand for more information about each role</b></summary>

- The **[content team lead](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Content-Team-Leads'-Guide)** is the chapter owner and responsible for setting the scope of the chapter and managing contributors' day-to-day progress.
- **[Authors](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Authors'-Guide)** are subject matter experts and lead the content direction for each chapter. Chapters typically have one or two authors. Authors are responsible for planning the outline of the chapter, analyzing stats and trends, and writing the annual report.
- **[Reviewers](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Reviewers'-Guide)** are also subject matter experts and assist authors with technical reviews during the planning, analyzing, and writing phases.
- **[Analysts](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Analysts'-Guide)** are responsible for researching the stats and trends used throughout the Almanac. Analysts work closely with authors and reviewers during the planning phase to give direction on the types of stats that are possible from the dataset, and during the analyzing/writing phases to ensure that the stats are used correctly.
- **[Editors](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Editors'-Guide)** are technical writers who have a penchant for both technical and non-technical content correctness. Editors have a mastery of the English language and work closely with authors to help wordsmith content and ensure that everything fits together as a cohesive unit.
- The **[section coordinator](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Section-Leads'-Guide)** is the overall owner for all chapters within a section like "User Experience" or "Page Content" and helps to keep each chapter on schedule.

_Note: The time commitment for each role varies by the chapter's scope and complexity as well as the number of contributors._

For an overview of how the roles work together at each phase of the project, see the [Chapter Lifecycle](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Chapter-Lifecycle) doc.
</details>

### [**0. Form the content team**](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Chapter-Lifecycle#0-create-content-team)
+ [x] **May 31**: The content team has at least one author, reviewer, and analyst

### [**1. Plan content**](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Chapter-Lifecycle#1-plan-content)
+ [x] **June 15** The content team has completed the chapter outline in the [draft doc][~google-doc]

### [**2. Gather data**](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Chapter-Lifecycle#2-gather-data)
+ [x] **June 30**: Analysts have added all necessary [custom metrics](https://github.com/HTTPArchive/legacy.httparchive.org/blob/master/custom_metrics/README.md) and drafted a PR ([example](https://github.com/HTTPArchive/almanac.httparchive.org/pull/1087)) to track query progress
- **July 1 - 31**: HTTP Archive runs the July crawl

### [**3. Validate results**](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Chapter-Lifecycle#3-validate-results)
+ [x] **September 30**: Analysts have queried all metrics and saved the output to the [results sheet][~google-sheets]

### [**4. Draft content**](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Chapter-Lifecycle#4-draft-content)
+ [x] **October 31**: The content team has written, reviewed, and edited the chapter in the [doc][~google-doc]

### [**5. Publication**](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Chapter-Lifecycle#5-publication)
+ [x] **November 15**: The completed chapter and all required [metadata](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Authors'-Guide#metadata-to-add-at-the-top-of-your-chapters) and [figures](https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Figures-Guide) are converted to markdown and submitted to [GitHub][~chapter-markdown]
- **December 1**: Target launch date 🚀

## Chapter resources

Refer to these 2021 CDN resources throughout the content creation process:

[📄 Google Docs][~google-doc] for outlining and drafting content
[🔍 SQL files][~sql-files] for committing the queries used during analysis
[📊 Google Sheets][~google-sheets] for saving the results of queries
[📝 Markdown file][~chapter-markdown] for publishing content and managing public metadata

[~google-doc]: https://docs.google.com/document/d/1OJBK345mSYsWxm0dhvMVgGY0pfUqC8jUjmvgwK8I3Fw/edit?usp=sharing
[~sql-files]: https://github.com/HTTPArchive/almanac.httparchive.org/tree/main/sql/2021/cdn
[~google-sheets]: https://docs.google.com/spreadsheets/d/1DL7Pn1vbBwYmQZ5JPAjD69oCOvUidbuoNvdrw--j00U/edit?usp=sharing
[~chapter-markdown]: https://github.com/HTTPArchive/almanac.httparchive.org/tree/main/src/content/en/2021/cdn.md

### PR DESCRIPTION
**Pull request by shantsis**
Progress on #2525
Closes #2159 

Stage link: https://20211116t174953-dot-webalmanac.uk.r.appspot.com/en/2021/cdn

Basic edits and some TODOs for the author to review

### CODE REVIEW

File: src/content/en/2021/cdn.md
Comment (shantsis) [inline src/content/en/2021/cdn.md:299]: @tunetheweb thoughts on the line above where it says (no pun intended ;))?

File: src/content/en/2021/cdn.md
Comment (shantsis) [inline src/content/en/2021/cdn.md:65]: @Navaneeth-akam

File: src/content/en/2021/cdn.md
Comment (shantsis) [inline src/content/en/2021/cdn.md:116]: @Navaneeth-akam

File: src/content/en/2021/cdn.md
Comment (shantsis) [inline src/content/en/2021/cdn.md:144]: @Navaneeth-akam

File: src/content/en/2021/cdn.md
Comment (shantsis) [inline src/content/en/2021/cdn.md:187]: @Navaneeth-akam

File: src/content/en/2021/cdn.md
Comment (Navaneeth-akam) [reply src/content/en/2021/cdn.md:65]: I would prefer it to be in paragraph style :)

File: src/content/en/2021/cdn.md
Comment (Navaneeth-akam) [reply src/content/en/2021/cdn.md:116]: Ahhh....a bit of typo here. It should have been "we will look at this data a bit different below"

File: src/content/en/2021/cdn.md
Comment (Navaneeth-akam) [reply src/content/en/2021/cdn.md:144]: yes. its a typo. But your suggestion is good. We can put 2.7 million as a big number and highlight it. That will get some attention

File: src/content/en/2021/cdn.md
Comment (Navaneeth-akam) [reply src/content/en/2021/cdn.md:187]: Yes. This one had mixed opinion from the reviewers. They felt having it as bullet points is better. But i do feel putting them in a sentence makes more sense. We can go ahead and put them in a sentence

File: src/content/en/2021/cdn.md
Comment (Navaneeth-akam) [reply src/content/en/2021/cdn.md:299]: Not sure about the policy for this. I'm other either way

**Comment by Navaneeth-akam** [comment]
@shantsis : I have added my comments to your queries. Please review and let me know if they make sense

**Review by Navaneeth-akam — COMMENTED** [review]
Added my comments where ever i was tagged

File: src/content/en/2021/cdn.md
Comment (shantsis) [reply src/content/en/2021/cdn.md:144]: I haven't added images before, are you able to commit an update here to add it in?

Comment (Navaneeth-akam) [reply src/content/en/2021/cdn.md:144]: I don't have a big number figure as such. I was thinking more along the lines of increased font size, making them bold and centering it between the paragraph. Does this make sense?

Comment (Navaneeth-akam) [reply src/content/en/2021/cdn.md]: Else, we can just make the 2.7 million in bold as a highlight if a figure is not doable

Comment (shantsis) [reply src/content/en/2021/cdn.md]: there's a reference for it here https://github.com/HTTPArchive/almanac.httparchive.org/wiki/Figures-Guide#call-out-figures
maybe @tunetheweb can help?

**Comment by shantsis** [comment]
> @shantsis : I have added my comments to your queries. Please review and let me know if they make sense

Thanks for the quick response @Navaneeth-akam! I pushed one update to close off most of the comments. There's 2 left that I'll wait for @tunetheweb to weigh in on.

File: src/content/en/2021/cdn.md
Comment (rviscomi) [reply src/content/en/2021/cdn.md]: @Navaneeth-akam if it exists in the [results sheet](https://docs.google.com/spreadsheets/d/1DL7Pn1vbBwYmQZ5JPAjD69oCOvUidbuoNvdrw--j00U/edit) it's easy to make it into a big number figure.

In the cdn_usage_by_site_rank tab I see 2,012,646 mobile "requests" (should this be "pages"?) that use a CDN in the ALL ranking. Would 2M be more accurate or did you get 2.7M from a different query?

Comment (Navaneeth-akam) [reply src/content/en/2021/cdn.md]: i calculated 2.7 million looking at the numbers in the results. You will see that for "Top 10 million site", 27% use CDN. Which translates to 2.7 million sites out of 10 million use CDN.

**Comment by Navaneeth-akam** [comment]
@shantsis I added the big number figure for "2.7 million". You can review it to see if it fits in correctly.
@rviscomi : Please review the figure markup for this. I have linked the google sheet ID and sql query to the same one used for "site popularity ranking" query

File: src/content/en/2021/cdn.md
Comment (rviscomi) [reply src/content/en/2021/cdn.md]: Oh ok there are only 7,491,736 pages in the "all" rank, which is why we avoid calling it 10M.

Comment (Navaneeth-akam) [reply src/content/en/2021/cdn.md]: Ahh....ok. Gotcha. Then may be we can take out that big number thing. Because we are now quoting top 10 million sites, so trying to take 27% of 7,491,736 will not be equal to 27% of 10 million, Thus it might be confusing for readers how we came up with this number. Thoughts?

Comment (tunetheweb) [reply src/content/en/2021/cdn.md]: Why not just put 27% in a big number?

Comment (Navaneeth-akam) [reply src/content/en/2021/cdn.md]: That thought did cross my mind after i commented :) We can do that. Works for me. 👍🏼

**Review by Navaneeth-akam — COMMENTED** [review]
Updated big number figure

File: src/content/en/2021/cdn.md
Comment (rviscomi) [inline src/content/en/2021/cdn.md:136]:
```suggestion
Ranking the websites based on their popularity (sourced from Google's [Chrome UX Report](./methodology#chrome-ux-report)) in the web and then checking for their CDN usage, the top 1,000 contribute to the highest usage of CDN. The top websites are owned by companies like Google and Amazon, who contribute to much of the internet traffic we see today, so it's no surprise that these names make it to the list of [top CDN providers](#top-CDN-providers) in the next section. This also backs the fact about the benefits CDNs bring to the table when operating at scale and having the ability to scale further if needed.
```

File: src/content/en/2021/cdn.md
Comment (tunetheweb) [inline src/content/en/2021/cdn.md:145]: @Navaneeth-akam we avoid using 10 million number (since it's not). Also SQL file looked incorrect.

```suggestion
{{ figure_markup(
  caption="Sites using CDN.",
  content="27%",
  classes="big-number",
  sheets_gid="414030881",
  sql_file="cdn_usage_by_site_rank.sql"
)
}}
```

File: src/content/en/2021/cdn.md
Comment (rviscomi) [inline src/content/en/2021/cdn.md:143]:
```suggestion
  sql_file="cdn_usage_by_site_rank.sql"
```

File: src/content/en/2021/cdn.md
Comment (rviscomi) [inline src/content/en/2021/cdn.md:139]: @Navaneeth-akam should we also specify that this is usage of a CDN for **hosting** the site (serving the HTML document) to distinguish from CDN usage of other 1P and 3P resources?

```suggestion
  caption="Percent of websites using a CDN.",
```

File: src/content/en/2021/cdn.md
Comment (rviscomi) [inline src/content/en/2021/cdn.md:148]:
```suggestion
The CDN adoption rate falls below 50% when we look at the top 100,000 websites but the rate of reduction slows down beyond this. For the full data set (which is 6.2 million sites on desktop and 7.5 million on mobile), 27% of these websites use CDN. When you translate that percentage into real number, that's 2.0 million mobile websites using CDN! It's not such a small number when you look at it this way.
```

**Review by rviscomi — CHANGES_REQUESTED** [review]
@shantsis could you also see if there are any other places in the chapter to apply this kind of feedback?

**Review by Navaneeth-akam — COMMENTED** [review]
I have changed the big number to 61% to reflect the CDN usage by top 1000 sites. I understand the total of 10 million is not correct, but i'm guessing the top 1000 sites should be correct

File: src/content/en/2021/cdn.md
Comment (Navaneeth-akam) [reply src/content/en/2021/cdn.md]: Understood. I did a change to the big number to show the CDN usage for top 1000 instead of the 10 million number

**Comment by shantsis** [comment]
> @shantsis could you also see if there are any other places in the chapter to apply this kind of feedback?

which kind of feedback? Sorry I don't see any link

**Comment by rviscomi** [comment]
>which kind of feedback? Sorry I don't see any link

My editing style is a bit more "invasive" 😁
For example, see my [review](https://github.com/HTTPArchive/almanac.httparchive.org/pull/2584#discussion_r756312019) feedback. Also checking the `sql_file` for other figures to make sure they're valid files.

**Comment by shantsis** [comment]
> > which kind of feedback? Sorry I don't see any link
> 
> My editing style is a bit more "invasive" 😁 For example, see my [review](https://github.com/HTTPArchive/almanac.httparchive.org/pull/2584#discussion_r756312019) feedback. Also checking the `sql_file` for other figures to make sure they're valid files.

Oh I see thanks! I'm not familiar with this topic so that's why my editing was more focused on checking links/grammar. Good call on sql_file, I'll start paying attention to that.

File: src/content/en/2021/cdn.md
Comment (tunetheweb) [reply src/content/en/2021/cdn.md:293]: I say remove if you don't mind.

**Comment by Navaneeth-akam** [comment]
@shantsis let me know when the editing is completed. I'll look into any comment you might have added and have that addressed ASAP

**Review by Navaneeth-akam — COMMENTED** [review]
Removed the emoji and TODO comment under HTTP/2 adoption section

**Comment by shantsis** [comment]
@rviscomi don't have any further comments on my end

File: src/content/en/2021/cdn.md
Comment (rviscomi) [inline src/content/en/2021/cdn.md:139]:
```suggestion
  caption="Percent of top 1,000 websites using a CDN.",
```

File: src/content/en/2021/cdn.md
Comment (rviscomi) [inline src/content/en/2021/cdn.md:367]:
```suggestion
On both desktop and mobile platforms, we see that CDNs are delivering twice as many text assets in Brotli, unlike domains delivered from origin.  From the [CDN Adoption](#cdn-adoption) section covered earlier, 73% of the domains serving HTMLs are on CDNs and these all  benefit from the Brotli compression. By offloading the computational load of compressing a text asset in the Brotli format to CDNs, website owners need not invest resources for hosting infrastructure.
```

**Review by rviscomi — APPROVED** [review]

**Review by Navaneeth-akam — COMMENTED** [review]
Updating org info in author bio section

File: src/content/en/2021/cdn.md
Comment (tunetheweb) [inline src/content/en/2021/cdn.md:65]: I toned this down a litte. Hope you don't mind @Navaneeth-akam

File: src/content/en/2021/cdn.md
Comment (tunetheweb) [inline src/content/en/2021/cdn.md:76]: I added this sentence to explain why (I think) you choose not to measure these. Please review and let me know if you want to change.

File: src/content/en/2021/cdn.md
Comment (tunetheweb) [inline src/content/en/2021/cdn.md:208]: @Navaneeth-akam does this need clarification? I was a bit confused.

File: src/content/en/2021/cdn.md
Comment (tunetheweb) [inline src/content/en/2021/cdn.md:286]: Tweaked this slightly to introduce HTTP/3. Let me know if you want to change.

**Review by tunetheweb — CHANGES_REQUESTED** [review]
@Navaneeth-akam I did another run through and made some more minor edits, and also normalised all the stats to 1 decimal place consistently.

I've one question for you, and a few changes of mine you should review to confirm you're happy with some changes I made.

After that I think we're done and good to merge!

File: src/content/en/2021/cdn.md
Comment (Navaneeth-akam) [reply src/content/en/2021/cdn.md:286]: Works for me Barry. I'm good with these tweaks

File: src/content/en/2021/cdn.md
Comment (Navaneeth-akam) [reply src/content/en/2021/cdn.md:65]: Yup....looks good.

File: src/content/en/2021/cdn.md
Comment (Navaneeth-akam) [reply src/content/en/2021/cdn.md:76]: Yes, the addition makes sense. Adds more color to the statement.

Comment (tunetheweb) [reply src/content/en/2021/cdn.md:76]: Just spotted a typo and repeated use of the phrase “our crawl” so let’s use below tweak instead:

```suggestion
While some of these could be measured with HTTP Archive dataset, and others by using the CrUX dataset, the limitations of our methodology and the use of multiple CDNs by some sites, will be difficult to measure and so could be incorrectly attributed. For these reasons, we have decided not to measure these statistics in this chapter.
```

File: src/content/en/2021/cdn.md
Comment (Navaneeth-akam) [reply src/content/en/2021/cdn.md:208]: Long tail content management is not something generic CDN's are good at by default. A purpose built CDN can be designed to manage this a bit better. As for geographical closeness, generic CDN will be deployed based on certain assessment but their geographical closeness is relative to Origin. Purpose built CDN will have a focused end user set where they will need deployment of server, which differs a bit from generic CDN

**Comment by Navaneeth-akam** [comment]
> @Navaneeth-akam I did another run through and made some more minor edits, and also normalised all the stats to 1 decimal place consistently.
> 
> I've one question for you, and a few changes of mine you should review to confirm you're happy with some changes I made.
> 
> After that I think we're done and good to merge!

Hi Barry, i responded to your comments. Few of your suggestions made sense and we should be good to incorporate them. I have clarified on the purpose built CDN part. Let me know if it makes sense

File: src/content/en/2021/cdn.md
Comment (tunetheweb) [reply src/content/en/2021/cdn.md:208]: The long tails makes sense (missed that to be honest), and guess that similarly applies to high input/output. I guess for geographical closeness I considered all CDNs to have good global presence but that’s maybe not the case?

Anyway let’s remove the TODO if happy with the wording.

```suggestion
```

**Review by tunetheweb — APPROVED** [review]
This is good to merge! Great to see the CDN chapter back for 2021 after a year sabbatical!

**Pull request merged**
